Framework/python/ldmxcfg.py

```diff
@@ -36,17 +37,111 @@ def __init__(self, instanceName, className):
 
         import LDMX.Framework.histogram as h
-        self.histograms.append(h.histogram1D(name, xlabel, bins, xmin, xmax))
-        return self
+        theBinEdges = bins
+        if xmin is not None and xmax is not None :
+            theBinEdges = h.uniform_binning(bins,xmin,xmax)
+
+        self.histograms.append(h.histogram(name, xlabel,theBinEdges))
+
+    def build2DHistogram(self, name, 
+            xlabel = 'X Axis', xbins = 1, xmin = None, xmax = None, 
+            ylabel = 'Y Axis', ybins = 1, ymin = None, ymax = None) :
+        """Create a 2D histogram
+
+        If {x,y}min or {x,y}max are not provided, {x,y}bins is assumed
+        to be the bin edges on the {x,y}-axis. If they are both provided,
+        {x,y}-bins is assumed to be the number of bins on the {x,y}-axis.
+
+        Parameters
+        ----------
+        name : str
+            variable name of histogram
+        xlabel : str
+            title of x-axis of histogram
+        xbins : int OR list of floats
+            Number of bins on x-axis OR list of bin edges on x-axis
+        xmin : float
+            Minimum edge of bins on x-axis
+        xmax : float
+            Maximum edge of bins on x-axis
+        ylabel : str
+            title of y-axis of histogram
+        ybins : int OR list of floats
+            Number of bins on y-axis OR list of bin edges on y-axis
+        ymin : float
+            Minimum edge of bins on y-axis
+        ymay : float
+            Mayimum edge of bins on y-axis
+
+        See Also
+        --------
+        LDMX.Framework.histogram.histogram : histogram configuration object
+
+        Examples
+        --------
+
+        When doing all uniform binning, you can specify the arguments by position.
+            myProcessor.build2DHistogram( 'dummy' ,
+                'My X Axis' , 20 , 0. , 1. ,
+                'My Y Axis' , 60 , 0. , 10. )
+
+        When using variable binning, you have to use the parameter names.
+            myProcessor.build2DHistogram( 'dummy2' ,
+                xlabel='My X Axis', xbins=[0.,1.,2.],
+                ylabel='My Y Axis', ybins=60, ymin=0., ymax=10. )
+        """
+
+        import LDMX.Framework.histogram as h
+        theBinEdgesX = xbins
+        if xmin is not None and xmax is not None :
+            theBinEdgesX = h.uniform_binning(xbins,xmin,xmax)
+
+        theBinEdgesY = ybins
+        if ymin is not None and ymax is not None :
+            theBinEdgesY = h.uniform_binning(ybins,ymin,ymax)
+
+        self.histograms.append(
+                h.histogram(name, xlabel,theBinEdgesX, ylabel,theBinEdgesY)
+                )
+
+class Producer(EventProcessor):
+    """A producer object.
+
+    This object contains the parameters that are necessary for a ldmx::Producer to be configured.
+
+    See Also
+    --------
+    LDMX.Framwork.ldmxcfg.EventProcessor : base class
+    """
+
+    def __init__(self, instanceName, className):
+        super().__init__(instanceName,className)
```

Comment:
This is the line that is not compliant with Python2. I could remove the base class and endure some code copying if we still wanted to support Python2.